utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/data_streams/DSMContextCarrier.java

```diff
@@ -0,0 +1,35 @@
+package com.datadoghq.system_tests.springboot.data_streams;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import datadog.trace.api.experimental.DataStreamsContextCarrier;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DSMContextCarrier implements DataStreamsContextCarrier {
+    @JsonProperty
+    private Map<String, Object> data;
+
+    public DSMContextCarrier() {
+      this.data = new ConcurrentHashMap<>();
+    }
+
+    public DSMContextCarrier(Map<String, Object> data) {
+      this.data = data;
+    }
+
+    @Override
+    public synchronized Set<Map.Entry<String, Object>> entries() {
```

Comment:
there is already a concurrent hashmap 
Perhaps having a synchronized method is not needed here